### ISSUE TITLE
Ssr doesn't work if use getContext in entries.ts 's defineRouter/getComponent

### ISSUE THREAD

**Issue by EasonYou** [dai-shi/waku#104]
# What's the issue

I use ssr with ``routes`` and ``getContext``

As the code show, it doesn't work if use ``getContext`` in defineRouter/getComponent.

https://github.com/EasonYou/waku-example/blob/71308df89396c7fa6b5c9ce54ff46d8fafcdf7aa/src/entries.ts#L1-L21

<img width="766" alt="image" src="https://github.com/dai-shi/waku/assets/16911502/557ea340-7b6e-4037-a040-cf6c2b5530bb">

# What do I find
The error tells that ctx was empty, it cause by the ``getComponent`` so that waku can't get element by ``getSsrConfig`` because of the component import error. Then the ssr will not work. But no error on the client side, just like a csr app.

https://github.com/dai-shi/waku/blob/5af1c3461df9ac02472775cd16ae6d382d1726c9/src/router/server.ts#L155-L171

I don't know if it is a purposeful design that user can't use ``getSsrConfig`` or it is a bug.

**Comment by dai-shi** [comment]
Thanks for reporting!
Does it fix or change something if you comment out L162 in `src/router/server.ts`?

**Comment by EasonYou** [comment]
No, if you commont out L162, all the ``ssrConfig`` will be got in the ssr middleware(see the code below L99). It cause that all the request wiil be handled as a ssr page.
I have time to try to fix it, maybe I can create a pr later.
https://github.com/dai-shi/waku/blob/5af1c3461df9ac02472775cd16ae6d382d1726c9/src/lib/middleware/ssr.ts#L98-L104

**Comment by dai-shi** [comment]
Oh, yeah, that's right..

I think we should avoid using `getComponent` in `getSsrConfig`.

**Comment by EasonYou** [comment]
And I find that ``index`` is the default route, If I visit the ``index`` route, it will be render the ``index`` component as child component.

<img width="517" alt="image" src="https://github.com/dai-shi/waku/assets/16911502/6e91c594-4cf6-42a3-a383-7fab0220fa2c">

btw, is that any issue or discussion about routes design?

**Comment by EasonYou** [comment]
``Index`` component is default as the routes' entry, but I find that if I visit ``index`` path, it will render the ``index`` components as child.
![image](https://github.com/dai-shi/waku/assets/16911502/01246be8-e2e1-4ec4-9041-24bb610227b1)

**Comment by dai-shi** [comment]
Oh, I wasn't aware that could happen. (So, it's a bug...)

Right, I think how we handle `index` is a bit weird compared to community standard.
I would like to fix/improve the file name convention and support `[slug].tsx` too.

Let's tackle them separately.
